### PR TITLE
enhance: Set segment.maxSize param to 1024M

### PR DESCRIPTION
**Pull request by xige-16**
issue: #25639 
/kind improvement

When the number of vector columns increases, the number of rows per segment will decrease. In order to reduce the impact on vector indexing performance, it is necessary to increase the segment max limit.

 If a collection has multiple vector fields with memory and disk indices on different vector fields, the size limit after segment compaction is the minimum of segment.maxSize and segment.diskSegmentMaxSize.

Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
rerun ut

File: internal/datacoord/compaction_trigger.go [hunk 1]
```diff
@@ -315,31 +318,44 @@ func (t *compactionTrigger) updateSegmentMaxSize(segments []*SegmentInfo) (bool,
...
+	})
+	vectorFieldsWithDiskIndex := lo.Filter(vectorFields, func(field *schemapb.FieldSchema, _ int) bool {
+		if indexType, ok := fieldIndexTypes[field.FieldID]; ok {
+			return indexparamcheck.IsDiskIndex(indexType)
+		}
+		return false
+	})
+
+	allDiskIndex := len(vectorFields) == len(vectorFieldsWithDiskIndex)
+	if allDiskIndex {
```
Comment (longjiquan) [inline internal/datacoord/compaction_trigger.go:339]: Maybe we should re-calculate the max segment size as long as there are any disk indexes.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/datacoord/compaction_trigger.go (hunk 1 above)
Comment (xige-16) [reply internal/datacoord/compaction_trigger.go:339]: May result in larger memory index

**Pull request merged**